src/proposals.rs

```diff
@@ -231,7 +231,7 @@ async fn poll_approved_proposals(
                 info!("ignoring rejected/failed proposal of ID {}", proposal_id);
                 if pending_proposals {
                     pending_skip_count += 1;
-                    pending_already_processed.insert(proposal_id);
+                    pending_finalized.insert(proposal_id);
```

Comment:
It's a bit below this line, I can't tag it directly, but I think if we run into an "unspecified" proposal status we'll still need to make sure we skip it.